frame/ibc/src/lib.rs

```diff
@@ -0,0 +1,313 @@
+	>;
+
+	#[pallet::storage]
+	/// (port_identifier, channel_identifier) => Sequence
+	pub type NextSequenceRecv<T: Config> = StorageDoubleMap<
+		_,
+		Blake2_128Concat,
+		Vec<u8>,
+		Blake2_128Concat,
+		Vec<u8>,
+		Vec<u8>,
+		ValueQuery,
+	>;
+
+	#[pallet::storage]
+	/// (port_identifier, channel_identifier) = Sequence
+	pub type NextSequenceAck<T: Config> = StorageDoubleMap<
+		_,
+		Blake2_128Concat,
+		Vec<u8>,
+		Blake2_128Concat,
+		Vec<u8>,
+		Vec<u8>,
+		ValueQuery,
+	>;
+
+	#[pallet::storage]
+	/// (port_identifier, channel_identifier, sequence) => Hash
+	pub type Acknowledgements<T: Config> =
+		StorageMap<_, Blake2_128Concat, (Vec<u8>, Vec<u8>, Vec<u8>), Vec<u8>, ValueQuery>;
+
+	#[pallet::storage]
+	/// clientId => ClientType
+	pub type Clients<T: Config> =
+		CountedStorageMap<_, Blake2_128Concat, Vec<u8>, Vec<u8>, ValueQuery>;
+
+	#[pallet::storage]
+	/// client_id => Connection id
+	pub type ConnectionClient<T: Config> =
+		StorageMap<_, Blake2_128Concat, Vec<u8>, Vec<u8>, ValueQuery>;
+
+	#[pallet::storage]
+	/// (port_id, channel_id, sequence) => receipt
+	pub type PacketReceipt<T: Config> =
+		StorageMap<_, Blake2_128Concat, (Vec<u8>, Vec<u8>, Vec<u8>), Vec<u8>, ValueQuery>;
+
+	#[pallet::storage]
+	/// (port_id, channel_id, sequence) => hash
+	pub type PacketCommitment<T: Config> =
+		StorageMap<_, Blake2_128Concat, (Vec<u8>, Vec<u8>, Vec<u8>), Vec<u8>, ValueQuery>;
+
+	#[pallet::storage]
+	/// height => IbcConsensusState
+	pub type CommitmentRoot<T: Config> =
+		StorageMap<_, Blake2_128Concat, Vec<u8>, IbcConsensusState, ValueQuery>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T> {
+		/// Processed incoming ibc messages
+		ProcessedIBCMessages,
+	}
+	/// Errors inform users that something went wrong.
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Error processing ibc messages
+		ProcessingError,
+		/// Error decoding some type
+		DecodingError,
+		/// Error encoding some type
+		EncodingError,
+		/// Error inserting a value in trie
+		TrieInsertError,
+		/// Error generating trie proof
+		ProofGenerationError,
+		/// Client consensus state not found for height
+		ConsensusStateNotFound,
+		/// Client state not found
+		ClientStateNotFound,
+		/// Error constructing packet
+		SendPacketError,
+		/// Other forms of errors
+		Other,
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T>
+	where
+		u32: From<<T as frame_system::Config>::BlockNumber>,
+	{
+		fn on_finalize(_n: BlockNumberFor<T>) {
+			let root = Pallet::<T>::extract_ibc_state_root();
+			if let Ok(root) = root {
+				let height = crate::impls::host_height::<T>();
+				let height = ibc::Height::new(0, height);
+				let height = height.encode_vec().unwrap_or_default();
+				let timestamp = T::TimeProvider::now().as_nanos().saturated_into::<u64>();
+				let ibc_cs = IbcConsensusState { timestamp, root: root.clone() };
+				CommitmentRoot::<T>::insert(height, ibc_cs.clone());
```

Comment:
don't think we use this value anywhere

Reply:
We'll need it here https://github.com/ComposableFi/composable/pull/829/files#diff-15d5711034ee5ceacf10a76d9f210d95de421a30f7fec36497517888be809135R151 after the consensus state  for  a beefy light client makes its way into ibc-rs, I need a way to prune it though, we only need recent ones, probably dating back a 100 blocks at most.